pennylane/devices/device_constructor.py

```diff
@@ -271,6 +278,12 @@ def _safe_specifier_set(version_str):
         # Once the device is constructed, we set its custom expansion function if
         # any custom decompositions were specified.
         if custom_decomps is not None:
+            warnings.warn(
+                """The ``custom_decomps`` keyword argument to ``qml.device`` has been deprecated and will be removed 
+                in v0.44. Instead, with ``qml.decomposition.enable_graph()``, define custom decomposition rules as a
+                quantum function, and register it with ``qml.add_decomps``.""",
```

Comment:
```suggestion
                """The ``custom_decomps`` keyword argument to ``qml.device`` has been deprecated and will be removed 
                in v0.44. Instead, use the graph-based system with ``qml.decomposition.enable_graph()``, and define new decomposition rules as
                quantum functions with registered resources. See https://docs.pennylane.ai/en/stable/code/qml_decomposition.html for more details.""",
```

This will have to be run through black 😅. 

@andrijapau @astralcai what's the standard for having actual links (urls) in warning messages 🤔 ?

Reply:
Good question, they are typically introduced by hyperlinking words,
```
see `Custom Devices <https://docs.pennylane.ai/projects/catalyst/en/stable/dev/custom_devices.html>`_.
```
Maybe we can do something similar here?

Reply:
Done!